src/Systems/Systems.jl

```diff
@@ -15,107 +15,156 @@ export AbstractSystem,
 Base.next(NDInput::ConstantNonDeterministicInput, state) = InputState(NDInput.U)
 Base.done(NDInput::ConstantNonDeterministicInput, state) = false
 Base.eltype(::Type{ConstantNonDeterministicInput}) = Type{InputState}
 Base.length(NDInput::ConstantNonDeterministicInput) = 1
 
-import Base: *
 
 function *(M::AbstractMatrix{Float64}, NDInput::ConstantNonDeterministicInput)
-    return ConstantNonDeterministicInput(M * start(NDInput).sf)
+    return ConstantNonDeterministicInput(M * start(NDInput).set)
 end
 
+
 """
-Type that represents a time-varying non-deterministic input.
+Type that represents a time-varying nondeterministic input.
+
+The iteration over this set is such that its `state` is a tuple
+(`set`, `index`), where `set` is the value of the input, represented as an array
+of `LazySet`s, and `index` counts the number of times this iterator was called.
+Its length corresponds to the number of elements in the given array. The index
+of the input state increases from 1 and corresponds at each time to the array
+index in the input array.
+
+### Fields
 
-The iteration over this set is such that its `state` is a tuple (sf, index), where
-sf is the value of the input, represented as a LazySet set, and `index` counts
-the number of times this iterator was called. Its length corresponds to the number
-of elements in the given array of support functions. The index of the input state
-increases from 1 and corresponds at each time to the array index in the input array U.
+- `U` -- array containing `LazySet`s
 """
 struct TimeVaryingNonDeterministicInput <: NonDeterministicInput
     # input sequence
-    U::Array{<:LazySet, 1}
+    U::Vector{<:LazySet}
 
-    TimeVaryingNonDeterministicInput(U) = new(U)
+    TimeVaryingNonDeterministicInput(U::Vector{<:LazySet}) = new(U)
 end
 
+
 Base.start(NDInput::TimeVaryingNonDeterministicInput) = InputState(NDInput.U[1])
 Base.next(NDInput::TimeVaryingNonDeterministicInput, state) = InputState(NDInput.U[state.index+1], state.index+1)
 Base.done(NDInput::TimeVaryingNonDeterministicInput, state) = state.index > length(NDInput.U)
 Base.eltype(::Type{TimeVaryingNonDeterministicInput}) = Type{InputState}
 Base.length(NDInput::TimeVaryingNonDeterministicInput) = length(NDInput.U)
 
+
+#=
+Systems
+=#
+
+
 """
 Abstract type representing a system of affine ODEs.
 """
 abstract type AbstractSystem end
 
+
 """
-Type that represents an continous-time affine system with non-deterministic inputs,
+Type that represents a continous-time affine system with nondeterministic inputs,
 
-x'(t) = Ax(t) + u(t),
+``x'(t) = Ax(t) + u(t)``,
 
 where:
 
-- A is a square matrix
-- x(0) ∈ X0
-- u(t) ∈ U(t), where U(t) is a piecewise-constant set-valued function defined
-  over [t1, t1+δ], ... , [tN, tN+δ]
+- ``A`` is a square matrix
+- ``x(0) ∈ X0``
+- ``u(t) ∈ U(t)``, where ``U(t)`` is a piecewise-constant set-valued function
+  defined over ``[t1, t1+δ], ... , [tN, tN+δ]`` for some δ
```

Comment:
the fact that there is no `delta`as a field of this type, but it's still mentioned in the docstring, is a bit misleading, i acknowledge.. maybe we should just say (quoting decoPaper):

> where ``U(\cdot)`` is a piecewise-constant set-valued function, i.e. we consider a possibly time-varying discrete sequence ``\{U(k)\}_k``